frontend/lib/ui/dashboard/dashboard_body/artefacts_list_view/headers.dart

```diff
@@ -11,21 +11,61 @@ class _Headers extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    final uri = AppRoutes.uriFromContext(context);
+    final sortBy = uri.queryParameters[CommonQueryParameters.sortBy];
+    final sortDirection =
+        uri.queryParameters[CommonQueryParameters.sortDirection];
+
     return SizedBox(
       height: 56,
       child: Row(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
         children: columnsMetaData
             .map(
               (data) => Expanded(
                 flex: data.flex,
-                child: Text(
-                  data.name,
-                  style: Theme.of(context).textTheme.titleLarge,
+                child: InkWell(
+                  onTap: () => _handleFilterTap(context, uri, data),
+                  child: Row(
+                    children: [
+                      Flexible(
+                        child: Text(
+                          data.name,
+                          style: Theme.of(context).textTheme.titleLarge,
+                        ),
+                      ),
+                      if (sortBy == data.queryParam.name)
+                        if (sortDirection == SortDirection.asc.name)
+                          const Icon(Icons.arrow_upward)
+                        else
+                          const Icon(Icons.arrow_downward),
+                    ],
+                  ),
                 ),
               ),
             )
             .toList(),
       ),
     );
   }
+
+  void _handleFilterTap(
+    BuildContext context,
+    Uri pageUri,
+    ColumnMetadata columnData,
+  ) {
+    final queryParameters = pageUri.queryParameters;
+    final sortBy = columnData.queryParam.name;
+    final direction = queryParameters[CommonQueryParameters.sortBy] == sortBy &&
+            queryParameters[CommonQueryParameters.sortDirection] ==
+                SortDirection.asc.name
+        ? SortDirection.desc.name
+        : SortDirection.asc.name;
```

Comment:
Can we simplify the logic here a bit, it seems like we are trying to handle the following: If the header we are pressing is different than the currently chosen one use ascending order, otherwise use descending sort, but it seems a bit difficult to understand written all in one line?

Reply:
Thanks for letting me know that this was difficult to read. You are half right. We only set descending order if both the chosen header is the same as the current one, and the existing direction is ascending. I've updated the code to make this clearer